src/Mod/Sketcher/App/SketchObject.cpp

```diff
@@ -2375,21 +2376,20 @@ int SketchObject::delConstraintOnPoint(int geoId, PointPos posId, bool onlyCoinc
 
     findReplacement();
 
-    auto performCoincidenceChecksOrChanges = [&](auto& constr) -> std::optional<bool> {
-        if (constr->Type != Sketcher::Coincident) {
-            if (onlyCoincident) {
-                return true;
-            }
-            return std::nullopt;
-        }
-        bool firstSucceeded = transferToReplacement(constr->First, constr->FirstPos);
-        bool secondSucceeded = transferToReplacement(constr->Second, constr->SecondPos);
-        return (firstSucceeded || secondSucceeded)
-            && (constr->First != constr->Second || constr->FirstPos != constr->SecondPos);
+    auto performCoincidenceChecksOrChanges = [&](auto& constr) -> bool {
+        return replaceGeoId != GeoEnum::GeoUndef
+            && !constr->involvesGeoIdAndPosId(replaceGeoId, replacePosId)
+            && (transferToReplacement(constr->First, constr->FirstPos)
+                || transferToReplacement(constr->Second, constr->SecondPos));
     };

```

Comment:
This is a bit hard to read, how about splitting it up a bit?
```c++
    auto performCoincidenceChecksOrChanges = [&](auto& constr) -> bool {
        if (replaceGeoId == GeoEnum::GeoUndef) {
            return false;
        }
        if (constr->involvesGeoIdAndPosId(replaceGeoId, replacePosId)) {
            return false;
        }

        return transferToReplacement(constr->First, constr->FirstPos)
                || transferToReplacement(constr->Second, constr->SecondPos);
    };
```

Reply:
I thought the conditions being one-line would keep it clear enough. `transferToReplacement()` can be safely replaced by `constr->substituteGeoIdAndPosId()` here (all relevant checks are already done), except that it is a non-returning function.

There's also a bit of "cognitive complexity golf" going on here: the `if` statements add a +2 instead of a +1 for the `&&`. I rationalized it by considering that the former unnecessarily lengthen the code and someone familiar with C++ would instantly know that the second part of `&&` is not computed if the first is false.

Any opinion on this respect?

Reply:
The problem isn't about if people will be able to figure it out or if people don't understand c++.
The problem is that it is a big blob of text that you need to tokenize and parse when reading it.

By splitting it up, you make it less cognitive complex as you just see:
* false if geoid is undef
* false if constrain involves the geo and pos
* true when either ... or ... 

Code is always easier to read when you write it than for someone else to read it.

Would be interesting what other people think here, it could just be me :)

Reply:
I agree with @hyarion here, conditions that mix `||` and &&` are really hard to understand and should be avoided. This blob of text is really hard to parse and understand what part does what. I'd probably even add a bit of comments to explain what is the reason for each condition as they are not obvious to the reader.

Reply:
I have changed the code according to the suggestion. If `substituteIndexAndPos()` returned bool, I would have preferred it to be:
```
auto performCoincidenceChecksOrChanges = [&](auto& constr) -> bool {
        return replaceGeoId != GeoEnum::GeoUndef
            && !constr->involvesGeoIdAndPosId(replaceGeoId, replacePosId)
            && substituteIndexAndPos(geoId, posId, replaceGeoId, replacePosId);
};
```
However, this is a pre-existing function so better not change it.

Reply:
This would have worked too

Reply:
OK marking as resolved until we get a better way to do this.